File: OneCodeTeam/Claim-aware-WCF-WebServicevs10-17866/[C#]-Claim-aware WCF WebService[vs10] (CSSharePointCallClaimsAwareWCF)/C#/CSSharePointCallClaimsAwareWCF/Service/CustomServiceHostFactory.cs
/****************************** Module Header ******************************\
* Module Name:    CustomServiceHostFactory.cs
* Project:        CSSharePointCallClaimsAwareWCF
* Copyright (c) Microsoft Corporation
*
* This is the Custom ServiceHostFactory Class
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\*****************************************************************************/

using System;
using System.ServiceModel;
using System.ServiceModel.Activation;
using Microsoft.SharePoint.Administration;
using Microsoft.SharePoint;

namespace CustomService
{
    internal sealed class CustomServiceHostFactory : ServiceHostFactory
    {
        public override ServiceHostBase CreateServiceHost(string constructorString, Uri[] baseAddresses)
        {
            ServiceHost serviceHost = new ServiceHost(typeof(CustomServiceApplication), baseAddresses);

            // Configure the service host for claims authentication
            serviceHost.Configure(SPServiceAuthenticationMode.Claims);
            return serviceHost;
        }
    }
}

File: OneCodeTeam/Claim-aware-WCF-WebServicevs10-17865/[VB]-Claim-aware WCF WebService[vs10] (VBSharePointCallClaimsAwareWCF)/VB/VBSharePointCallClaimsAwareWCF/Service/CustomServiceHostFactory.vb
'*************************** Module Header ******************************\
' Module Name:    CustomServiceHostFactory.vb
' Project:        VBSharePointCallClaimsAwareWCF
' Copyright (c) Microsoft Corporation
'
' This is the Custom ServiceHostFactory Class
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'\****************************************************************************


Imports System.ServiceModel
Imports System.ServiceModel.Activation
Imports Microsoft.SharePoint.Administration
Imports Microsoft.SharePoint

Namespace CustomService
    Friend NotInheritable Class CustomServiceHostFactory
        Inherits ServiceHostFactory
        Public Overrides Function CreateServiceHost(ByVal constructorString As String, ByVal baseAddresses As Uri()) As ServiceHostBase
            Dim serviceHost As New ServiceHost(GetType(CustomServiceApplication), baseAddresses)

            ' Configure the service host for claims authentication
            serviceHost.Configure(SPServiceAuthenticationMode.Claims)
            Return serviceHost
        End Function
    End Class
End Namespace
